src/snowcli/cli/streamlit/manager.py

```diff
@@ -38,6 +37,48 @@ def share(self, streamlit_name: str, to_role: str) -> SnowflakeCursor:
     def drop(self, streamlit_name: str) -> SnowflakeCursor:
         return self._execute_query(f"drop streamlit {streamlit_name}")
 
+    def _deploy_experimental(
+        self,
+        streamlit_name: str,
+        main_file: Path,
+        environment_file: Optional[Path] = None,
+        pages_dir: Optional[Path] = None,
+        stage_name: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        replace: Optional[bool] = False,
+    ):
+        database = self._conn.database.upper()  # type: ignore[attr-defined]
+        schema = self._conn.schema.upper()  # type: ignore[attr-defined]
+        embedded_stage_name = f"snow://streamlit/{database}.{schema}.{streamlit_name}"
+        root_location = f"{embedded_stage_name}/default_checkout"
+
+        replace_stmt = "OR REPLACE" if replace else ""
+
+        # TODO: Support from_stage
+        # from_stage_stmt = f"FROM_STAGE = '{stage_name}'" if stage_name else ""
+
+        use_warehouse_stmt = f"QUERY_WAREHOUSE = {warehouse}" if warehouse else ""
+
+        self._execute_query(
+            f"""
+            CREATE {replace_stmt} STREAMLIT {streamlit_name}
+            MAIN_FILE = '{main_file.name}'
+            {use_warehouse_stmt}
+            """
+        )
+
+        stage_manager = StageManager()
+
+        stage_manager.put(main_file, root_location, 4, True)
+
+        if environment_file and environment_file.exists():
+            stage_manager.put(environment_file, root_location, 4, True)
+
+        if pages_dir and pages_dir.exists():
+            stage_manager.put(pages_dir / "*.py", f"{root_location}/pages", 4, True)
+
+        return self.get_url(streamlit_name)
```

Comment:
We can probably extract this to command implementation to have:
```py
def deploy(...)
   manager = StreamlitManager(...)
   if experimental:
      manager.deploy_experimental(...)
   else :
      manager.deploy(...)
    
    url = manager.get_url(streamlit_name)
    return MessageResult(...)
```

WDYT?

Reply:
Yeah, that's not a bad idea. I'll take a swing at refactoring to something like that.

Reply:
Let me know what you think of the refactoring I did above

Reply:
Look awesome now! Thanks